### DISCUSSION TITLE
Unable to Find Sample Codes for Filament Android Native C++ API and Issues with Connecting to Native Android Window

### DISCUSSION THREAD

**Discussion by BhaveshDilipJ** [google/filament#7887] [General]
Hello,

I am currently using Filament as a rendering engine and working on Linux Ubuntu. I have successfully generated output on my Linux machine using Filament and C++. However, I am facing difficulties when attempting to produce the same output on Android using the C++ Native API.

Issues:

Sample Codes Missing:
I am unable to find sample codes inside the filament-main/android/filament-android folder. It seems the samples provided do not cover my use case for Android with Native C++.

Connecting Filament with Native Android Window:
On Linux, I use ::getNativeWindow(sdlWindow) to connect Filament with the native window. However, I am unsure how to achieve this on Android. There seems to be no direct equivalent or clear documentation on this process.

Specific Question:
Could you provide guidance or examples on how to connect Filament to a Native Android window or activity? Any sample code or references to relevant documentation would be greatly appreciated.

Steps Taken:
Generated output on Linux using Filament and C++ successfully.
Checked the filament-main/android/filament-android directory for samples but could not find relevant examples for my use case.
Searched the documentation and issues on the Filament GitHub page but could not find a solution.

Environment:
Operating System: Linux Ubuntu
Filament Version: [1.52.0]
Android Version: [30.0.3]
NDK Version: [26.1.10909125]

Additional Context:
Any additional tips or pointers to get started with integrating Filament on Android using C++ Native APIs would be very helpful.

Thank you for your assistance!

Best regards,
Bhavesh Joshi.

**Comment by poweifeng** [comment]
You can look at [UiHelper](https://github.com/google/filament/blob/main/android/filament-android/src/main/java/com/google/android/filament/android/UiHelper.java) to get an idea of how the java-side of Android gets set up (obtaining a surface that you can forward to engine etc.). Then you'd need to replicate that in native.  Lastly, see [android-samples](https://github.com/google/filament/tree/main/android/samples) for information on how to run the Android samples (albeit the entry points are all in java/kotlin with jni bindings to Filament).
